lib/matplotlib/axes/_axes.py

```diff
@@ -5871,6 +5871,40 @@ def hist(self, x, bins=10, range=None, normed=False, weights=None,
         .. plot:: mpl_examples/statistics/histogram_demo_features.py
 
         """
+        def _normalize_input(inp, ename='input'):
+            """Normalize 1 or 2d input into list of np.ndarray or
+            a single 2D np.ndarray.
```

Comment:
I was waiting for a clearer docstring here because the if-statements are opaque.